syntax = "proto3";

service NumStore {
	rpc SetNum(SetNumRequest) returns (SetNumResp);
        rpc Fact(FactRequest) returns (SetFactResp);
}

message SetNumRequest {
	string key = 1;
	int64 value = 2;
}

message SetNumResp {
	int64 total = 1;
}

message FactRequest {
	string key = 1;
}

message SetFactResp {
	int64 value = 1;
	bool hit = 2;
	string error = 3;
}
